File: Week 4/DemoDistributedObject/DemoDistributedObject/CObject.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DemoDistributedObject
{
    public abstract class CObject
    {
        protected int Handle;
        internal void Attach(int handle)
        {
            Handle = handle;
        }

        internal int Detach()
        {
            int handle = Handle;
            Handle = 0;

            return handle;
        }

    }
}

File: Week 4/Java code/Middleware/src/client/CObject.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package client;

/**
 *
 * @author chungnt
 */
public abstract class CObject {
    protected int handle;
    
    void attach(int handle){
        this.handle = handle;
    }
    
    int dettach(){
        int handle = this.handle;
        this.handle = 0;
        return handle;
    }
    
}
